tools/htdocs/components/20_VRForm.js

```diff
@@ -0,0 +1,447 @@
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+Ensembl.Panel.VRForm = Ensembl.Panel.ToolsForm.extend({
+
+  init: function() {
+    this.base();
+
+    this.resetSpecies(this.defaultSpecies);
+
+    this.consequencesData = this.params['consequences_data'];
+    delete this.params['consequences_data'];
+
+    this.exampleData = this.params['example_data'];
+    delete this.params['example_data'];
+
+    // this.autocompleteData = this.params['plugin_auto_values'];
+    // delete this.params['plugin_auto_values'];
+
+    var panel = this;
+
+    // Change the input value on click of the examples link
+    this.elLk.form.find('a._example_input').on('click', function(e) {
+      e.preventDefault();
+
+      var species = panel.elLk.form.find('input[name=species]:checked').val() || panel.elLk.form.find('input[name=species]').val();
+      var text    = panel.exampleData[species][this.rel];
+      if (typeof text === 'undefined' || !text.length) {
+        text = '';
+      }
+      text = text.replace(/\\n/g, "\n");
+
+      panel.elLk.dataField.val(text).trigger('focus');
+    });
+
+    // Preview button
+    this.elLk.previewButton = panel.elLk.form.find('[name=preview]').hide().on('click', function(e) {
+      e.preventDefault();
+      panel.enablePreviewButton(false);
+      panel.preview($(this).data('currentVal'), { left : e.pageX, top : e.pageY });
+    });
+
+    // Preview div
+    this.elLk.previewDiv = $('<div class="vep-preview-div">').appendTo($(document.body));
+
+    // show hide preview button acc to the text in the input field
+    this.elLk.dataField = this.elLk.form.find('textarea[name=text]').on({
+      'paste keyup click change focus scroll': function(e) {
+        panel.dataFieldInteraction(e.type);
+      }
+    });
+
+    // move preview button after the textarea
+    this.elLk.previewButton.appendTo(this.elLk.dataField.parent());
+
+    this.elLk.form.find('.plugin_enable').change(function() {
+
+      panel.elLk.form.find('.plugin-highlight').removeClass('plugin-highlight');
+
+      // find any sub-fields enabling this plugin shows
+      panel.elLk.form.find('._stt_' + this.name).addClass('plugin-highlight', 100, 'linear');
+    });
+
+    // also remove highlighting when option changes
+    this.elLk.form.find('.plugin_enable').each(function() {
+      panel.elLk.form.find('._stt_' + this.name).find(':input').change(function() {
+        panel.elLk.form.find('.plugin-highlight').removeClass('plugin-highlight');
+      });
+    });
+
+    this.editExisting();
+
+    // initialise the selectToToggle fields ie. show or hide the ones as needed
+    this.resetSelectToToggle();
+  },
+
+  dataFieldInteraction: function(eventType) {
```

Comment:
The whole function can be deleted, its related with the preview for VEP

Reply:
Commented